### ISSUE TITLE
Internal error with interaction point in a shared type signature

### ISSUE THREAD

**Issue by maxsnew** [agda/agda#7590]
Got an __IMPOSSIBLE__ on an incomplete program when using Agda version 2.7.0.1 . minimal example:
```
module Repro where

f g : ∀ x → ? ?
```

Gets me:
```
Location of the error: __IMPOSSIBLE__, called at src/full/Agda/TypeChecking/MetaVars.hs:476:11 in Agd-2.7.0.1-cd9573e7:Agda.TypeChecking.MetaVars% 
```

**Comment by AndrasKovacs** [comment]
Smaller example:

```
x y : (A : Set) → ? A
```
Alternatively:
```
x y : (A : Set)(B : A → Set) → B ?
```
The problem: there are two places where we look at the context of an already existing interaction point:
- In [Rules.LHS.Term.isType_](https://github.com/agda/agda/blob/3425ed5543d2e6f98094b834fedcdec3a2fb67a6/src/full/Agda/TypeChecking/Rules/Term.hs#L181), we check that the length of the current context is not smaller than the original context of the interaction point. If not, we fall back to creating a fresh interaction point.
- In [TypeChecking.MetaVars.newQuestionMark'](https://github.com/agda/agda/blob/3425ed5543d2e6f98094b834fedcdec3a2fb67a6/src/full/Agda/TypeChecking/MetaVars.hs#L461), we check if the original context is a prefix of the new one (by comparing abstract Name-s), and throw IMPOSSIBLE if that's not the case. 

In short, if a hole in a multiple-name type signature is checked with `isType_`, we don't crash, otherwise we do crash because the `Name`-s differ in the old and new contexts.

The original issue seems to be #707.

It seems to me that `isType_` and `newQuestionMark` should have the same behavior w.r.t. existing interaction points, whatever the behavior is. 

Changing `newQuestionMark` to match `isType_` would fix this, but I don't know if that's the best we could do here.

**Comment by andreasabel** [comment]
@AndrasKovacs : Do you want to try and fix this for 2.8.0?

**Comment by AndrasKovacs** [comment]
@andreasabel Yes.
